docs/performance.rst

```diff
@@ -0,0 +1,108 @@
+Performance
+===========
+
+This section contains an analysis of the overhead of running ``memray`` over different
```

Comment:
Typographic nit: you've consistently used \`\`memray\`\` here to refer to the name of the tool. I've tried to consistently use Memray as the name of the tool in all of our other docs pages, and I use \`\`memray\`\` only when I'm referring to the name of the importable module or the entry point script.